### ISSUE TITLE
main.cpp : stringstream flag clearance

### ISSUE THREAD

**Issue by kryv** [frib-high-level-controls/FLAME#2]
In main.cpp : line 184, 844, 1047 are

`str.str(line)`

where it is declared above like,
~~~
 std::string       line;
 std::stringstream str;
~~~

The stringstream object `str` (left hand side of `str.str(line)`) is used repeatedly without clearing any fail and eof flags.
As a result, the "str" was not overwrited and providing wrong calculation. 

Simply, we can solve this by `clear()`
~~~
str.str(line)
str.clear()  // add this line
~~~
`str.clear()` only resets the flag and the content `str(line)` is not deleted.
After `str.clear()`, the "str" is overwritable.

cf. http://stackoverflow.com/questions/2848087/how-to-clear-stringstream

Even if fix this bug, TLM_Lite and FLAME do not return the same transverse result,
but the reference particle energy (ionEk) history is completely same in LS1 section.
The transverse rms difference is  ~ 0.5 mm order in this time.

The transverse difference seems to be caused by wrong physics parameter inside the source code.
I'm going to continue bugfix.

**Comment by jbengtsson** [comment]
Below is the cross check for the phase-I.

TLM:

------------------------------------------------------------------------------------------------------------------------------------------------------------
calFribPropagate_MultiChargeStates
Ek =   5.00310e-01 Fy_abs=  -1.84773e-04
Ek =   5.02060e-01 Fy_abs=   2.29789e-02

 -7.8860000000e-04  1.0837100000e-05  1.3373430000e-02  6.6785340000e-06
-1.8477290000e-04  3.0999500000e-04  1.0000000000e+00

  7.3126670000e-03  1.4765650000e-05  3.4498490000e-03 -7.3976846900e-06
 2.2978860000e-02  2.0601000000e-03  1.0000000000e+00

 -1.3148719751e+00 -6.1411622599e-04 -7.3168243897e-01 -7.4661842888e-04
-2.3981901020e-03 -4.7275873408e-04  1.0000000000e+00

  1.2898546568e+00  5.2576860946e-04 -1.0781834290e-01 -8.7527544161e-05
-1.7446659046e-04 -4.8963879645e-05  0.0000000000e+00
  5.2576860946e-04  4.6323095062e-07 -1.8762087408e-04 -4.6135416622e-08
 4.1055348227e-08  1.8630565917e-08  0.0000000000e+00
 -1.0781834290e-01 -1.8762087408e-04  2.5417296760e+00  4.5070786513e-04
 2.2990389744e-04  1.1422604125e-04  0.0000000000e+00
 -8.7527544161e-05 -4.6135416622e-08  4.5070786513e-04  2.0980752246e-07
 1.6712357383e-07  7.0028121779e-08  0.0000000000e+00
 -1.7446659046e-04  4.1055348227e-08  2.2990389744e-04  1.6712357383e-07
 3.4570073319e-04  1.2937010003e-04  0.0000000000e+00
 -4.8963879645e-05  1.8630565917e-08  1.1422604125e-04  7.0028121779e-08
 1.2937010003e-04  5.1851103497e-05  0.0000000000e+00
  0.0000000000e+00  0.0000000000e+00  0.0000000000e+00  0.0000000000e+00
 0.0000000000e+00  0.0000000000e+00  0.0000000000e+00

 -1.9433786003e+00  3.4278187682e-04 -1.8746179484e+00 -7.8544844589e-04
 2.0808595171e-02 -3.0837818127e-03  1.0000000000e+00

  1.6617407569e+00 -1.1561293957e-04  4.2375967316e-01  5.2156789402e-06
-3.2654073343e-04 -1.6658632148e-04  0.0000000000e+00
 -1.1561293957e-04  2.0635559849e-07  1.7108084384e-06 -3.3149516381e-08
 1.5559377700e-07  8.0035719812e-08  0.0000000000e+00
  4.2375967316e-01  1.7108084384e-06  2.5993321065e+00 -3.1005696180e-05
-4.2384040459e-04 -2.0736703236e-04  0.0000000000e+00
  5.2156789402e-06 -3.3149516381e-08 -3.1005696180e-05  1.3142616031e-07
 1.4856230930e-07  7.0098541854e-08  0.0000000000e+00
 -3.2654073343e-04  1.5559377700e-07 -4.2384040459e-04  1.4856230930e-07
 7.5949774422e-04  3.9605913592e-04  0.0000000000e+00
 -1.6658632148e-04  8.0035719812e-08 -2.0736703236e-04  7.0098541854e-08
 3.9605913592e-04  2.0888918966e-04  0.0000000000e+00
  0.0000000000e+00  0.0000000000e+00  0.0000000000e+00  0.0000000000e+00
 0.0000000000e+00  0.0000000000e+00  0.0000000000e+00
------------------------------------------------------------------------------------------------------------------------------------------------------------

FLAME:
------------------------------------------------------------------------------------------------------------------------------------------------------------
InitLong:
  IonZ = 0.13866  IonEs [Mev/u] = 931.49432, IonEk [Mev/u] = 0.50000, IonW
[Mev/u] = 931.99432

InitLong: 0.01312 sec

InitLattice:
  IonZ = 0.13866  IonEs [Mev/u] = 931.49432, IonEk [Mev/u] = 0.50000, IonW
[Mev/u] = 931.99432
  IonZ = 0.14286  IonEs [Mev/u] = 931.49432, IonEk [Mev/u] = 0.50000, IonW
[Mev/u] = 931.99432

 s [m] = 103.415

 -1.3148719751e+00 -6.1411622599e-04 -7.3168243897e-01 -7.4661842888e-04
-2.3981901020e-03 -4.7275873408e-04  1.0000000000e+00

  1.2898546568e+00  5.2576860946e-04 -1.0781834290e-01 -8.7527544160e-05
-1.7446659046e-04 -4.8963879645e-05  0.0000000000e+00
  5.2576860946e-04  4.6323095062e-07 -1.8762087408e-04 -4.6135416622e-08
 4.1055348226e-08  1.8630565917e-08  0.0000000000e+00
 -1.0781834290e-01 -1.8762087408e-04  2.5417296760e+00  4.5070786513e-04
 2.2990389744e-04  1.1422604125e-04  0.0000000000e+00
 -8.7527544160e-05 -4.6135416622e-08  4.5070786513e-04  2.0980752246e-07
 1.6712357383e-07  7.0028121779e-08  0.0000000000e+00
 -1.7446659046e-04  4.1055348226e-08  2.2990389744e-04  1.6712357383e-07
 3.4570073319e-04  1.2937010003e-04  0.0000000000e+00
 -4.8963879645e-05  1.8630565917e-08  1.1422604125e-04  7.0028121779e-08
 1.2937010003e-04  5.1851103497e-05  0.0000000000e+00
  0.0000000000e+00  0.0000000000e+00  0.0000000000e+00  0.0000000000e+00
 0.0000000000e+00  0.0000000000e+00  0.0000000000e+00

 s [m] = 103.415

 -1.9433786003e+00  3.4278187682e-04 -1.8746179484e+00 -7.8544844589e-04
 2.0808595171e-02 -3.0837818127e-03  1.0000000000e+00

  1.6617407569e+00 -1.1561293956e-04  4.2375967316e-01  5.2156789402e-06
-3.2654073343e-04 -1.6658632148e-04  0.0000000000e+00
 -1.1561293956e-04  2.0635559849e-07  1.7108084385e-06 -3.3149516381e-08
 1.5559377700e-07  8.0035719812e-08  0.0000000000e+00
  4.2375967316e-01  1.7108084385e-06  2.5993321065e+00 -3.1005696178e-05
-4.2384040459e-04 -2.0736703236e-04  0.0000000000e+00
  5.2156789402e-06 -3.3149516381e-08 -3.1005696178e-05  1.3142616031e-07
 1.4856230930e-07  7.0098541854e-08  0.0000000000e+00
 -3.2654073343e-04  1.5559377700e-07 -4.2384040459e-04  1.4856230930e-07
 7.5949774422e-04  3.9605913592e-04  0.0000000000e+00
 -1.6658632148e-04  8.0035719812e-08 -2.0736703236e-04  7.0098541854e-08
 3.9605913592e-04  2.0888918966e-04  0.0000000000e+00
------------------------------------------------------------------------------------------------------------------------------------------------------------

In other words, they should agree to machine precision. For this particular
case, I ran with "EmitGrowth = false" (a small effect; but it leads to loss
of significant digits).

1. Clearly a bug (str).

2. Which physics parameter (inside the source code)?

Johan

**Comment by kryv** [comment]
I add the comparison results between TLM_Lite and FLAME (after above stringstream bugfix).
The longitudinal results (ionEk, RMS z) are the same without output accuracy.

![ek_cmp](https://cloud.githubusercontent.com/assets/11188515/14405314/00598e64-fe5a-11e5-837d-0da94f025cce.png)

![xrms_cmp](https://cloud.githubusercontent.com/assets/11188515/14405327/3c6927ca-fe5a-11e5-87c7-7d3938b5f3ce.png)

![zrms_cmp](https://cloud.githubusercontent.com/assets/11188515/14405330/5f4ff41c-fe5a-11e5-9d60-ca66ab51a1be.png)

**Comment by kryv** [comment]
Sorry, my input parameter : "Mpultipole level" was not the same as TLM input.
After this fix, I got the same result also in the transverse direction.

![xrms_cmp2all](https://cloud.githubusercontent.com/assets/11188515/14416965/f0f5cbfc-ff7d-11e5-9f9f-0924fb747a9c.png)
